iocore/utils/I_Machine.h

```diff
@@ -77,22 +71,22 @@ struct Machine {
       @note This must be called before called @c instance so that the
       singleton is not @em inadvertently default initialized.
   */
-  static self *init(char const *name     = nullptr, ///< Host name of the machine.
-                    sockaddr const *addr = nullptr  ///< Primary IP address of the machine.
+  static self_type *init(char const *name     = nullptr, ///< Host name of the machine.
+                         sockaddr const *addr = nullptr  ///< Primary IP address of the machine.
   );
   /// @return The global instance of this class.
-  static self *instance();
-  bool is_self(const char *name);
-  bool is_self(const char *name, int name_len);
-  bool is_self(const IpAddr *ipaddr);
+  static self_type *instance();
+  bool is_self(std::string_view name);
+  bool is_self(std::string const &name);
```

Comment:
This seems redundant because of:  https://en.cppreference.com/w/cpp/string/basic_string/operator_basic_string_view

Reply:
Unfortunately not - because the internal table is keyed by `std::string` it's more efficient to have an explicit overload for that type to avoid duplicating the original string.